src/plugins/components/avatar.ts

```diff
@@ -0,0 +1,717 @@
+    },
+    text: 'xl',
+    roundedOrInner: {
+      rounded: 'xl',
+      dotPosition: '0',
+    },
+    curvedOrInner: {
+      rounded: '3xl',
+      dotPosition: '0',
+    },
+    fullDotPosition: '1.5',
+  },
+  avatar3XL: {
+    size: '28',
+    badge: {
+      size: '10',
+      position: '1.5',
+    },
+    dot: {
+      size: '4',
+    },
+    text: '2xl',
+    roundedOrInner: {
+      rounded: 'xl',
+      dotPosition: '0',
+    },
+    curvedOrInner: {
+      rounded: '3xl',
+      dotPosition: '0',
+    },
+    fullDotPosition: '2',
+  },
+  avatar4XL: {
+    size: '32',
+    badge: {
+      size: '12',
+      position: '2',
+    },
+    dot: {
+      size: '4',
+    },
+    text: '3xl',
+    roundedOrInner: {
+      rounded: 'xl',
+      dotPosition: '0',
+    },
+    curvedOrInner: {
+      rounded: '3xl',
+      dotPosition: '0',
+    },
+    fullDotPosition: '2.5',
+  },
+  avatarFull: {
+    rounded: 'full',
+    avatarInnerRounded: 'full',
+    avatarBadgePosition: '0',
+  },
+}
+
+export default plugin.withOptions(
+  function (options: PluginOption) {
+    const { prefix } = defu(options, defaultPluginOptions)
+
+    return function ({ addComponents, theme }) {
+      const config = theme(
+        'shurikenUi.avatar'
+      ) satisfies typeof defaultAvatarConfig
+
+      addComponents({
+        [`.${prefix}-avatar`]: {
+          [`@apply relative inline-flex shrink-0 items-center justify-center outline-none`]:
+            {},
+
+          [`.${prefix}-avatar-inner`]: {
+            [`@apply flex h-${config.avatarInner.size} w-${config.avatarInner.size} items-center justify-center overflow-hidden text-center transition-all duration-${config.avatarInner.duration}`]:
+              {},
+          },
+          [`.${prefix}-avatar-badge`]: {
+            [`@apply dark:bg-${config.avatarBadge.bgDark} absolute z-10 block overflow-hidden rounded-${config.avatarBadge.rounded} bg-${config.avatarBadge.bg}`]:
+              {},
+
+            [`.badge-img`]: {
+              [`@apply relative h-${config.avatarBadge.img.size} w-${config.avatarBadge.img.size} scale-90 rounded-${config.avatarBadge.img.rounded} object-cover`]:
+                {},
+            },
+          },
+          [`.${prefix}-avatar-img`]: {
+            [`@apply max-h-${config.avatarImg.maxSize} max-w-${config.avatarImg.maxSize} object-cover shadow-${config.avatarImg.shadow} dark:border-${config.avatarImg.borderDark}`]:
+              {},
+          },
+          [`.${prefix}-avatar-text`]: {
+            [`@apply font-${config.avatarText.font} text-center font-${config.avatarText.fontSize} uppercase`]:
+              {},
+          },
+          [`.${prefix}-avatar-dot`]: {
+            [`@apply dark:border-${config.avatarDot.borderDark} absolute block rounded-${config.avatarDot.rounded} border border-${config.avatarDot.border}`]:
+              {},
+            [`&.${prefix}-dot-primary`]: {
+              [`@apply ${config.avatarDot.primary}`]: {},
```

Comment:
classes prefix are missing on all the file

```suggestion
              [`@apply bg-${config.avatarDot.primary}`]: {},
```